test/e2e_node/eviction_test.go

```diff
@@ -736,8 +803,10 @@ func verifyEvictionOrdering(ctx context.Context, f *framework.Framework, testSpe
 			}
 		}
 		gomega.Expect(priorityPod).NotTo(gomega.BeNil())
-		gomega.Expect(priorityPod.Status.Phase).ToNot(gomega.Equal(v1.PodSucceeded),
-			fmt.Sprintf("pod: %s succeeded unexpectedly", priorityPod.Name))
+		if !isPodExpectedToSucceed(&priorityPod) {
+			gomega.Expect(priorityPod.Status.Phase).ToNot(gomega.Equal(v1.PodSucceeded),
+				fmt.Sprintf("pod: %s succeeded unexpectedly", priorityPod.Name))
+		}
```

Comment:
Is it OK if the pod is terminated with a failure in this case? Or should it be running?

Reply:
Yes, the expected behavior would be to finish with a failure (for pods expected to be evicted) or to continue running (for pods that are expected to run forever). No other pod besides the red ones with `sleepAfterExecuting` equal to `true`, should succeed.